scripts/argocd-store-token.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+#
+# Uses the ArgoCD credentials to create a secret in the cluster. The credentials
+# are stored in a "env file" format sourced by this script, and later on using
+# "kubectl" to create the secret.
+#
+
+shopt -s inherit_errexit
+set -Eeu -o pipefail
+
+# Target secret name, to be created with ArgoCD credentials.
+declare -r SECRET_NAME="${SECRET_NAME:-rhtap-argocd-integration}"
+# Secret's namespace.
+declare -r NAMESPACE="${NAMESPACE:-}"
+
+# Environment file to store the ArgoCD crednetials.
+declare -r ARGOCD_ENV_FILE="${ARGOCD_ENV_FILE:-/rhtap/argocd/env}"
+
+fail() {
+    echo "# [ERROR] ${*}" >&2
+    exit 1
+}
+
+# Waits for the environment file to be available.
+wait_for_env_file() {
+    echo "# Waiting for '${ARGOCD_ENV_FILE}' to be available..."
+    for i in {1..30}; do
+        [[ -r "${ARGOCD_ENV_FILE}" ]] &&
+            return 0
+
+        wait=$((i * 15))
```

Comment:
The step feels extra large. If my calculations are correct the max sleep time is 6975s, which seems overkill.

Reply:
Before it happens the Helm Chart installation times out will kick in, but it's definitely extra long.